### PR TITLE
👷‍♀️FIX:  resource update context input

### PR DESCRIPTION
**Pull request by kenjinp**
Simply makes the context input for updates match the context type used elsewhere in Resource

ALSO provides a fix for the resource path to make sure the ID is URL encoded

### CODE REVIEW

File: nexus-sdk-js/src/Resource/index.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ export default class Resource<T = {}> {
     );
     this.resourceURL = `/resources/${this.orgLabel}/${
       this.projectLabel
-    }/resource/${this.id}`;
+    }/resource/${encodeURIComponent(this.id)}`;
```
Comment (julienmachon) [inline nexus-sdk-js/src/Resource/index.ts]: Can you add a wee test to check if the id gets encoded properly? :kissing:

File: nexus-sdk-js/src/Resource/index.ts [hunk 2]
```diff
@@ -121,15 +121,15 @@ export default class Resource<T = {}> {
...
+    context?: Context;
     [field: string]: any;
   }): Promise<Resource> {
     return Resource.updateSelf(
       this.resourceURL,
       this.rev,
       {
         context,
-        data,
+        ...data,
```
Comment (julienmachon) [inline nexus-sdk-js/src/Resource/index.ts]: if this didn't break any tests, that means it isn't tested!

File: nexus-sdk-js/src/Resource/utils.ts [hunk 3]
```diff
@@ -157,7 +155,7 @@ export async function updateResource(
     context,
     ...data
   }: {
-    context: { [field: string]: string };
+    context: Context;
```
Comment (julienmachon) [inline nexus-sdk-js/src/Resource/utils.ts]: Context is optional for `updateSelfResource`, why is it not here?

File: nexus-sdk-js/src/Resource/index.ts (hunk 2 above)
Comment (kenjinp) [reply nexus-sdk-js/src/Resource/index.ts]: I've added a nice test

**Pull request merged**